compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -5065,36 +5071,154 @@ private void reorderArguments(BLangInvocation iExpr) {
         }
 
         BInvokableSymbol invokableSymbol = (BInvokableSymbol) symbol;
+
+        List<BLangExpression> restArgs = iExpr.restArgs;
+        int originalRequiredArgCount = iExpr.requiredArgs.size();
+
+        // Constructs used when the vararg provides args for required/defaultable params.
+        BLangExpression varargRef = null;
+        BLangBlockStmt blockStmt = null;
+
+        if (!iExpr.restArgs.isEmpty() &&
+                restArgs.get(restArgs.size() - 1).getKind() == NodeKind.REST_ARGS_EXPR &&
```

Comment:
`restArgs.get(restArgs.size() - 1)` 
I think we can extract this or related logic to a variable. This is used multiple times. This will simplify the code increasing readability.
L5083, L5088, L5098, L5115 (condition)